src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,112 @@
+package duke.parser;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.exception.ErrorMessage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a parser that deals with parsing user input and returning the correct command.
+ */
+public class Parser {
+    /**
+     * Parses the input string and returns the matching Command object that executes the command.
+     *
+     * @param input String representing user input.
+     * @return Command object.
+     * @throws DukeException if the input is not a valid command.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        if (input.isBlank()) {
+            throw new DukeException(ErrorMessage.MESSAGE_INVALID_INPUT);
```

Comment:
Good use of the `ErrorMessage` class! Now I'm considering implementing that in mine as well 😄